tests/test_invenio_access_permissions.py

```diff
@@ -61,33 +62,99 @@ def test_invenio_access_permission_for_users(app):
+        db.session.add(user_can_argument_dummy)
 
         db.session.add(ActionUsers(action='open', user=user_can_all))
         db.session.add(ActionUsers(action='open', user=user_can_open))
 
         db.session.add(ActionUsers(action='read', user=user_can_all))
         db.session.add(ActionUsers(action='read', user=user_can_read))
+
+        db.session.add(ActionUsers(action='not_logged', user=user_can_all))
+        db.session.add(ActionUsers(action='not_logged'))
+
+        db.session.add(ActionUsers(action='argument1',
+                                   user=user_can_argument_all))
+        db.session.add(ActionUsers(action='argument1',
+                                   argument='dummy',
+                                   user=user_can_argument_dummy))
+        db.session.add(ActionUsers(action='argument2',
+                                   argument='other',
+                                   user=user_can_argument_all))
         db.session.commit()
 
         permission_open = DynamicPermission(ActionNeed('open'))
         permission_read = DynamicPermission(ActionNeed('read'))
+        permission_not_logged = DynamicPermission(ActionNeed('not_logged'))
+        permission_argument1 = DynamicPermission(ActionNeed('argument1'))
+        permission_argument1_dummy = DynamicPermission(
+            ParameterizedActionNeed('argument1', 'dummy'))
+        permission_argument1_other = DynamicPermission(
+            ParameterizedActionNeed('argument1', 'other'))
+        permission_argument2 = DynamicPermission(ActionNeed('argument2'))
+        permission_argument2_dummy = DynamicPermission(
+            ParameterizedActionNeed('argument2', 'dummy'))
+        permission_argument2_other = DynamicPermission(
+            ParameterizedActionNeed('argument2', 'other'))
 
         identity_all = FakeIdentity(UserNeed(user_can_all.id))
         identity_read = FakeIdentity(UserNeed(user_can_read.id))
         identity_open = FakeIdentity(UserNeed(user_can_open.id))
+        identity_unknown = AnonymousIdentity()
+        identity_argument_all = FakeIdentity(
+            UserNeed(user_can_argument_all.id))
+        identity_argument_dummy = FakeIdentity(
+            UserNeed(user_can_argument_dummy.id))
 
+        # global permissions
         assert permission_open.allows(identity_all)
         assert permission_read.allows(identity_all)
+        assert permission_not_logged.allows(identity_all)
 
         assert permission_open.allows(identity_open)
         assert not permission_read.allows(identity_open)
+        assert not permission_not_logged.allows(identity_open)
 
         assert not permission_open.allows(identity_read)
         assert permission_read.allows(identity_read)
+        assert not permission_not_logged.allows(identity_read)
+
+        assert not permission_open.allows(identity_unknown)
+        assert not permission_read.allows(identity_unknown)
+        assert permission_not_logged.allows(identity_unknown)
```

Comment:
Here, we are in the case of rule 1 in the description of this PR. So, anybody, including not logged users, should access to this action.